src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStoragePublisher.java

```diff
@@ -143,10 +139,12 @@ public WAStoragePublisher(final String storageAccName,
 
     @DataBoundConstructor
     public WAStoragePublisher(final String storageCredentialId, final String filesPath,
-                              final String containerName) {
+                              final StorageType storageType) {
```

Comment:
This also has backward compatibility issue, let's change back to a flattened structure.